include/boost/redis/impl/sentinel_utils.hpp

```diff
@@ -0,0 +1,277 @@
+//
+// Copyright (c) 2025 Marcelo Zimbres Silva (mzimbres@gmail.com),
+// Ruben Perez Hidalgo (rubenperez038 at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#ifndef BOOST_REDIS_SENTINEL_UTILS_HPP
+#define BOOST_REDIS_SENTINEL_UTILS_HPP
+
+#include <boost/redis/config.hpp>
+#include <boost/redis/error.hpp>
+#include <boost/redis/resp3/node.hpp>
+#include <boost/redis/resp3/type.hpp>
+
+#include <boost/assert.hpp>
+#include <boost/core/ignore_unused.hpp>
+#include <boost/core/span.hpp>
+#include <boost/system/error_code.hpp>
+
+#include <cstddef>
+#include <string>
+#include <string_view>
+#include <utility>
+#include <vector>
+
+namespace boost::redis::detail {
+
+// Returns true if Sentinel should be used
+inline bool use_sentinel(const config& cfg) { return !cfg.sentinel.addresses.empty(); }
+
+// Composes the request to send to Sentinel modifying cfg.sentinel.setup
+inline void compose_sentinel_request(config& cfg)
+{
+   // These commands should go after the user-supplied setup, as this might involve authentication.
+   // We ask for the master even when connecting to replicas to correctly detect when the master doesn't exist
+   cfg.sentinel.setup.push("SENTINEL", "GET-MASTER-ADDR-BY-NAME", cfg.sentinel.master_name);
+   if (cfg.sentinel.server_role == role::replica)
+      cfg.sentinel.setup.push("SENTINEL", "REPLICAS", cfg.sentinel.master_name);
+   cfg.sentinel.setup.push("SENTINEL", "SENTINELS", cfg.sentinel.master_name);
+
+   // Note that we don't care about request flags because this is a one-time request
+}
+
+// Parses a list of replicas or sentinels
+inline system::error_code parse_server_list(
```

Comment:
- Can we move this to .ipp?
- Can we take a `std::vector<node>` instead and work with `.at()` function instead of iterators. We don't need performance here so I would prefer to be on the safer side.